gitapp/lib/user_details/details_page.dart

```diff
@@ -1,36 +1,69 @@
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 import 'package:gitapp/constants/constants.dart';
-import 'package:gitapp/user_details/user_details_model.dart';
 import 'package:gitapp/user_details/user_avatar_widget.dart';
 import 'package:gitapp/user_details/user_info_widget.dart';
-import 'package:gitapp/user_details/user_repo_model.dart';
 import 'package:gitapp/user_details/user_repos_list_widget.dart';
-import 'package:gitapp/user_details/user_details_presenter.dart';
+import 'package:gitapp/user_details/user_details_api_repository.dart';
 
-class DetailScreen extends StatelessWidget {
+Future<ReposList> getReposList(int userId) async => await getReposListFromAPI(userId);
+Future<Details> getDetails(int userId) async => await getInfoFromAPI(userId);
 
-  DetailScreen({required this.user});
-  final String user;
+class DetailScreen extends StatefulWidget {
+
+  DetailScreen({required this.userId});
+  final int userId;
 
   @override
-  Widget build(BuildContext context) {
+  _DetailScreenState createState() => _DetailScreenState();
+}
 
-    UserDetails mock = getMockedInfo(user);
-    List<UserRepo> repo = getMockedRepos(user);
+class _DetailScreenState extends State<DetailScreen> {
 
+  @override
+  void initState() {
+    super.initState();
+    getDetails(widget.userId);
+    getReposList(widget.userId);
+  }
+
+  @override
+  Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
         title: Text("User details"),
       ),
-      body: Column(
-        children: [
-          SizedBox(height: 20.0),
-          UserAvatarWidget(username: mock.username, avatarUrl: mock.avatarUrl),
-          UserInfoWidget(followersCount: mock.followersCount, repositoriesCount: mock.repositoriesCount, country: mock.country),
-          Divider(height: 35.0, color: yyellow, indent: 30.0, endIndent: 30.0),
-          UserReposWidget(listRepos: repo),
-        ]
+      body: SingleChildScrollView(
+        child: FutureBuilder(
+          future: Future.wait([getDetails(widget.userId), getReposList(widget.userId)]),
+          builder: (BuildContext context, AsyncSnapshot<dynamic> snapshot) {
+            if(snapshot.hasData) {
+              return Column(
+                  children: [
+                    SizedBox(height: 20.0),
+                    UserAvatarWidget(username: snapshot.data[0].userDetails.username, avatarUrl: snapshot.data[0].userDetails.avatarUrl),
+                    UserInfoWidget(followersCount: snapshot.data[0].userDetails.followersCount, repositoriesCount: snapshot.data[0].userDetails.repositoriesCount),
+                    Divider(height: 35.0, color: yyellow, indent: 30.0, endIndent: 30.0),
+                    UserReposWidget(userId: widget.userId, listRepos: snapshot.data[1].listRepos)
+                  ]
+              );
+            } else if(snapshot.hasError) {
+              throw Exception('Failed to fetch data from API.');
```

Comment:
Throwing an exception is a good idea? 
The app will crash when there will be a fail ?